src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -84,14 +84,14 @@ public static bool CanBeInitializedInline(CSharpType type, Constant? defaultValu
         {
              Debug.Assert(defaultValue.HasValue);
 
-            if (type.IsFrameworkType && type.FrameworkType == typeof(string))
+            if (type.IsString())
             {
                 return true;
             }
 
-            if (TypeFactory.IsStruct(type) && defaultValue.Value.Value != null)
+            if (IsStruct(type) && defaultValue.Value.Value != null)
             {
-                return false;
+                return defaultValue.Value.IsNewInstanceSentinel;
```

Comment:
Is this correct?

Reply:
Yes. If it is a sentinel, a struct can be initialized in parameter definition.

Reply:
Semantically though, we end up having incorrectly initialized enum value, but nothing prevents user from doing that manually anyway.